File: src/SpartanFitness.Contracts/Users/UserSavesResponse.cs
using SpartanFitness.Contracts.Exercises;
using SpartanFitness.Contracts.MuscleGroups;
using SpartanFitness.Contracts.Muscles;
using SpartanFitness.Contracts.Workouts;

namespace SpartanFitness.Contracts.Users;

public record UserSavesResponse(
  List<MuscleResponse> Muscles,
  List<MuscleGroupResponse> MuscleGroups,
  List<ExerciseResponse> Exercises,
  List<WorkoutResponse> Workouts);

File: src/SpartanFitness.ClientApp/src/types/responses/UserSavesResponse.ts
import Exercise from "../domain/Exercise";
import Muscle from "../domain/Muscle";
import MuscleGroup from "../domain/MuscleGroup";
import Workout from "../domain/Workout";

type UserSavesResponse = {
  exercises: Exercise[];
  workouts: Workout[];
  muscles: Muscle[];
  muscleGroups: MuscleGroup[];
};

export default UserSavesResponse;
